### PR TITLE
Feat/db queries

### PR DESCRIPTION
**Pull request by Blaradox**
created two functions that modify the database. One function will add orders to the database, and the other will update that order with a supplied delivery status and date in the 'YYYY-MM-DD' format. For now I will be ignoring adding products to each order, as that is not relevant to the business question and I want to focus first on data generation.

### CODE REVIEW

File: database/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    INSERT INTO orders
+    (user_id, shipping_address, delivery_status, created_at)
+    VALUES
+    ($1, $2, 'created', $3)
+  `;
+  const currDate = new Date().toISOString().slice(0, 10);
+  const queryArgs = [userId, shippingAddress, currDate];
+  return new Promise((resolve, reject) => {
+    client.query(queryStr, queryArgs, (err, res) => {
+      if (err) {
```
Comment (eugenesoo) [inline database/index.js]: not a big deal, but i guess ternary operators would do here as well?

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+const { Client } = require('pg');
+const Promise = require('bluebird');
+
+const client = new Client({
+  user: process.env.LENA_ORDERS_DB_USER || 'lena',
+  host: process.env.LENA_ORDERS_DB_HOST || 'localhost',
+  database: process.env.LENA_ORDERS_DB_data || 'transactions',
```
Comment (leungleoqin) [inline database/index.js]: LENA_ORDERS_DB_data should be all capitalized

File: database/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+client.connect();
+
+const createOrder = (userId, shippingAddress) => {
+  const queryStr = `
+    INSERT INTO orders
+    (user_id, shipping_address, delivery_status, created_at)
+    VALUES
+    ($1, $2, 'created', $3)
+  `;
+  const currDate = new Date().toISOString().slice(0, 10);
```
Comment (leungleoqin) [inline database/index.js]: Not a good idea to hardcode the slice index - even thought ISOStrings have consistent formats. Can either use a library to format it or use native getYear, getMonth, getDate functions.

File: database/index.js (hunk 1 above)
Comment (leungleoqin) [reply database/index.js]: I think this is good. Probably shouldn't use ternary operators to make function calls.

File: database/index.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  const queryArgs = [userId, shippingAddress, currDate];
+  return new Promise((resolve, reject) => {
```
Comment (leungleoqin) [inline database/index.js]: pg's query method already returns a Promise - no need to wrap it in another Promise. https://node-postgres.com/features/queries

File: database/index.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+const updateOrderStatus = (orderId, deliveryStatus, updatedAt) => {
+  const queryStr = `
+    UPDATE orders
+    SET
+    delivery_status = $2, updated_at = $3
+    WHERE order_id = $1
+  `;
+  const queryArgs = [orderId, deliveryStatus, updatedAt];
+  return new Promise((resolve, reject) => {
```
Comment (leungleoqin) [inline database/index.js]: Same as above - no need to wrap it in a Promise yourself

**Review by leungleoqin — CHANGES_REQUESTED** [review]

**Comment by Blaradox** [comment]
Thanks for all the comments @leungleoqin, @eugenesoo! I incorporated the changes you both mentioned.

File: database/index.js (hunk 2 above)
Comment (Blaradox) [reply database/index.js]: can't believe I missed that!

File: database/index.js (hunk 3 above)
Comment (Blaradox) [reply database/index.js]: Good idea, I found the npm package `yyyy-mm-dd`

File: database/index.js (hunk 4 above)
Comment (Blaradox) [reply database/index.js]: Thanks! I thought that this was the case, but wasn't sure how they handled errors, so I wanted to be explicit. However, I definitely made this change, it never makes sense to use redundant code.

**Pull request merged**